### PR TITLE
Removal and exchange duplicate test for privileged containers

### PR DESCRIPTION
**Pull request by horecoli**
## Description
This change removes the Kubescape implementation of test for privileged containers. The reason behind this decision is the absence of functionality to exclude containers from the test. Instead, it has been replaced with an in-house implementation of this test, which has been renamed from "privileged" to "privileged_containers".

## Issues:
Refs: #2115

## How has this been tested:
 - [x] Covered by existing integration testing
 - [ ] Added integration testing to cover
 - [ ] Verified all A/C passes
     * [ ] develop
     * [ ] master
     * [ ] tag/other branch
 - [ ] Test environment
    * [ ] Shared Packet K8s cluster
    * [ ] New Packet K8s cluster
    * [ ] Kind cluster
 - [ ] Have not tested

## Types of changes:
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
 - [ ] Documentation update

## Checklist:
**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. rescue block

**Issue**
- [ ] Tasks in issue are checked off

### CODE REVIEW

File: spec/workload/security_spec.cr [hunk 1]
```diff
@@ -4,35 +4,35 @@ require "../../src/tasks/utils/utils.cr"
...
 
-  it "'privileged' should pass with a non-privileged cnf", tags: ["privileged"]  do
+  it "'privileged_containers' should pass with a non-privileged cnf", tags: ["privileged"]  do
     begin
       result = ShellCmd.run_testsuite("cnf_setup cnf-config=sample-cnfs/sample-statefulset-cnf/cnf-testsuite.yml")
       Log.debug { result[:output] }
-      result = ShellCmd.run_testsuite("privileged verbose")
+      result = ShellCmd.run_testsuite("privileged_containers verbose")
       result[:status].success?.should be_true
       (/Found.*privileged containers.*coredns/ =~ result[:output]).should be_nil
```
Comment (kosstennbl) [inline spec/workload/security_spec.cr:13]: This output check doesn't seem right when comparing to test outputs, the test will always pass.

File: spec/workload/security_spec.cr [hunk 2]
```diff
@@ -4,35 +4,35 @@ require "../../src/tasks/utils/utils.cr"
 
 describe "Security" do
 
-  it "'privileged' should pass with a non-privileged cnf", tags: ["privileged"]  do
+  it "'privileged_containers' should pass with a non-privileged cnf", tags: ["privileged"]  do
```
Comment (kosstennbl) [inline spec/workload/security_spec.cr:7]: Maybe tag could be changed to privileged_containers? And same would be used for privilege escalation tests.

**Review by kosstennbl — CHANGES_REQUESTED** [review]
TEST_DOCUMENTATION.yml requires a small change.
utils_spec.cr:181-183, seems that prefix is also needed to be changed.

File: spec/workload/security_spec.cr (hunk 2 above)
Comment (horecoli) [reply spec/workload/security_spec.cr:7]: no, I don't think so, it was same also before without changing tag. Tag is probably used for all privileged tests including escalation.

File: spec/workload/security_spec.cr (hunk 1 above)
Comment (horecoli) [reply spec/workload/security_spec.cr:13]: I am not sure what you mean, I didn't change anything related to output, I just renamed test during its execution.

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -402,7 +402,7 @@ jobs:
         ./cnf-testsuite setup
         wget -O cnf-testsuite.yml https://raw.githubusercontent.com/cnti-testcatalog/testsuite/main/example-cnfs/coredns/cnf-testsuite.yml
         ./cnf-testsuite cnf_setup cnf-config=./cnf-testsuite.yml
-        LOG_LEVEL=info ./cnf-testsuite all ~compatibility ~resilience ~reasonable_startup_time ~reasonable_image_size ~platform ~privileged ~increase_capacity ~decrease_capacity ~install_script_helm ~helm_chart_valid ~helm_chart_published verbose
+        LOG_LEVEL=info ./cnf-testsuite all ~compatibility ~resilience ~reasonable_startup_time ~reasonable_image_size ~platform ~privileges ~increase_capacity ~decrease_capacity ~install_script_helm ~helm_chart_valid ~helm_chart_published verbose
```
Comment (kosstennbl) [inline .github/workflows/actions.yml:405]: I think task names are used there, not spec tags. Should be ~privileged_containers

File: .github/workflows/actions.yml [hunk 4]
```diff
@@ -477,7 +477,7 @@ jobs:
         ./cnf-testsuite setup
         wget -O cnf-testsuite.yml https://raw.githubusercontent.com/cnti-testcatalog/testsuite/main/example-cnfs/coredns/cnf-testsuite.yml
         ./cnf-testsuite cnf_setup cnf-config=./cnf-testsuite.yml
-        LOG_LEVEL=info ./cnf-testsuite all ~resilience ~compatibility ~pod_network_latency ~platform ~privileged ~increase_capacity ~decrease_capacity ~ip_addresses ~liveness ~readiness ~rolling_update ~rolling_downgrade ~rolling_version_change ~nodeport_not_used ~hostport_not_used ~hardcoded_ip_addresses_in_k8s_runtime_configuration ~install_script_helm ~helm_chart_valid ~helm_chart_published ~rollback ~secrets_used ~immutable_configmap verbose
+        LOG_LEVEL=info ./cnf-testsuite all ~resilience ~compatibility ~pod_network_latency ~platform ~privileges ~increase_capacity ~decrease_capacity ~ip_addresses ~liveness ~readiness ~rolling_update ~rolling_downgrade ~rolling_version_change ~nodeport_not_used ~hostport_not_used ~hardcoded_ip_addresses_in_k8s_runtime_configuration ~install_script_helm ~helm_chart_valid ~helm_chart_published ~rollback ~secrets_used ~immutable_configmap verbose
```
Comment (kosstennbl) [inline .github/workflows/actions.yml:480]: ditto

File: .github/workflows/actions.yml (hunk 3 above)
Comment (horecoli) [reply .github/workflows/actions.yml:405]: No, there was just "privileged" before, which was tag  and not a task name, that was the reason why I changed that only in last commit when I changed tag.

File: .github/workflows/actions.yml (hunk 4 above)
Comment (horecoli) [reply .github/workflows/actions.yml]: done

File: spec/workload/security_spec.cr (hunk 1 above)
Comment (horecoli) [reply spec/workload/security_spec.cr]: but yes, there was issue with regex before, which we didn't know about it. I fixed it in last commit. Thanks.

**Review by kosstennbl — APPROVED** [review]
LGTM

**Review by martin-mat — APPROVED** [review]
lgtm

**Review by rich-l — APPROVED** [review]
lgtm

**Pull request merged**